src/feature/hs/hs_cache.c

```diff
@@ -761,27 +785,41 @@ hs_cache_lookup_as_client(const ed25519_public_key_t *key)
   tor_assert(key);
 
   cached_desc = lookup_v3_desc_as_client(key->pubkey);
-  if (cached_desc) {
-    tor_assert(cached_desc->desc);
+  if (cached_desc && cached_desc->desc) {
```

Comment:
Perhaps let's explicilty document that this function will never return a non-decrypted descriptor.

Reply:
Fixup commit: e47973e15d